File: backend/models/QueueItem.js
const mongoose = require('mongoose');

const QueueItemSchema = new mongoose.Schema({
  appointmentId: { type: String, required: true },
  doctorId: { type: String, required: true },
  patientId: { type: String, required: true },
  patientName: { type: String, required: true },
  tokenNumber: { type: String, required: true },
  status: { type: String, enum: ['WAITING', 'IN_PROGRESS', 'COMPLETED', 'CANCELLED'], default: 'WAITING' },
  department: { type: String, default: null },
  date: { type: String, default: "" }, // Format: e.g. "Aug 17, 2026"
  isActive: { type: Boolean, default: true },
  timestamp: { type: Date, default: Date.now },
  estimatedWaitMinutes: { type: Number, default: 0 },
  consultationStartedAt: { type: Date },
  consultationCompletedAt: { type: Date },
  lastNotifiedWaitMinutes: { type: Number, default: 0 },
  lastNotifiedPosition: { type: Number, default: -1 }
});

module.exports = mongoose.model('QueueItem', QueueItemSchema);

File: app/src/main/java/com/example/medplus/model/QueueItem.kt
package com.example.medplus.model

import com.google.firebase.Timestamp
import com.google.gson.annotations.SerializedName

/**
 * Data model for an entry in the patient queue.
 */
data class QueueItem(
    @SerializedName("_id") val queueId: String = "",
    val appointmentId: String = "",
    val doctorId: String = "",
    val patientId: String = "",
    val patientName: String = "",
    val tokenNumber: String = "",
    val status: String = "WAITING", // WAITING, IN_PROGRESS, COMPLETED, CANCELLED
    val department: String? = null,
    val date: String = "",
    val isActive: Boolean = true,
    val timestamp: Timestamp = Timestamp.now(),
    val estimatedWaitMinutes: Int = 0,
    val consultationStartedAt: String? = null,
    val consultationCompletedAt: String? = null
)
